lib/auth/methods.go

```diff
@@ -347,20 +347,16 @@ func (s *Server) authenticateHeadless(ctx context.Context, req AuthenticateUserR
 	ctx, cancel := context.WithTimeout(ctx, defaults.CallbackTimeout)
 	defer cancel()
 
-	headlessAuthn := &types.HeadlessAuthentication{
-		ResourceHeader: types.ResourceHeader{
-			Metadata: types.Metadata{
-				Name: req.HeadlessAuthenticationID,
-			},
-		},
-		User:            req.Username,
-		PublicKey:       req.PublicKey,
-		ClientIpAddress: req.ClientMetadata.RemoteAddr,
-	}
-
 	// Headless Authentication should expire when the callback expires.
-	headlessAuthn.SetExpiry(s.clock.Now().Add(defaults.CallbackTimeout))
+	expires := s.clock.Now().Add(defaults.CallbackTimeout)
+	headlessAuthn, err := types.NewHeadlessAuthenticationStub(req.HeadlessAuthenticationID, expires)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 
+	headlessAuthn.User = req.Username
+	headlessAuthn.PublicKey = req.PublicKey
+	headlessAuthn.ClientIpAddress = req.ClientMetadata.RemoteAddr
```

Comment:
nit: `ClientIPAddress`

Even better would be to rename this field to `ClientLoginIP`. But I see this field wasn't added in this PR.

Reply:
As we are moving away from gogoproto, we are aiming to stick to the Protobuf best practices, including [using `snake_case` for proto fields](https://protobuf.dev/programming-guides/style/#message-field-names). There is no special case for names like `id` or `ip`, so unfortunately we get this result.